src/main/scala/Core/CSR.scala

```diff
@@ -119,22 +126,33 @@ class CSR(implicit val conf: FlexpretConfiguration) extends Module {
     CSR_W -> io.rw.data_in
   ))
 
+  // Check whether the write does not make any alterations; i.e. it is a fake write
+  // This is the case if the CSRRS (sets bits) or CSRRC (clears bits) provide
+  // bitmasks that are zero
+  // This allows non-priviledged mode for instructions that do not alter the CSRs
+  val fake_write = WireInit(false.B)
+  if (conf.privilegedMode) {
+    fake_write := MuxLookup(io.rw.csr_type, false.B, Array(
+      CSR_S -> (io.rw.data_in === 0.U),
+      CSR_C -> (io.rw.data_in === 0.U),
+      CSR_W -> false.B
+    ))
+  }
+
   // check permission (if privileged)
   val priv_fault = WireInit(false.B) // default value
   if (conf.privilegedMode) {
     val addr_read_only = io.rw.addr(11, 10) === 3.U // 0xC__ suggests read-only.
     val addr_no_priv = (io.rw.addr(9, 8) =/= 0.U) && (reg_prv(io.rw.thread) === 0.U(2.W))
-    when(io.rw.write && (addr_read_only || addr_no_priv)) {
+    when((io.rw.write && !fake_write) && (addr_read_only || addr_no_priv)) {
       priv_fault := true.B
     }
-    // TODO: read without permission?
```

Comment:
The above code implements this TODO, but FlexPRET is typically not compiled with `priviledgedMode`. I implemented this because I thought I needed `priviledgedMode` for context switches, but that was not the case after all